Source/Fuse.Vibration/Vibration.uno

```diff
@@ -25,6 +39,168 @@ namespace Fuse.Vibration
 		@}
 	}
 
+	[Require("Source.Include", "UIKit/UIKit.h")]
+	[Require("Xcode.Framework", "AudioToolbox")]
+	[ForeignInclude(Language.ObjC, "AudioToolbox/AudioToolbox.h")]
+	class IOSTapticFeedback
+	{
+		public static extern(iOS) void Perform(VibrationType style)
+		{
+			switch (style)
+			{
+				case VibrationType.Soft:
+					PerformSoft();
+					break;
+				case VibrationType.Rigid:
+					PerformRigid();
+					break;
+				case VibrationType.Light:
+					PerformLight();
+					break;
+				case VibrationType.Medium:
+					PerformMedium();
+					break;
+				case VibrationType.Heavy:
+					PerformHeavy();
+					break;
+				case VibrationType.Success:
+					PerformSuccess();
+					break;
+				case VibrationType.Warning:
+					PerformWarning();
+					break;
+				case VibrationType.Error:
+					PerformError();
+					break;
+				case VibrationType.Selection:
+					PerformSelection();
+					break;
+				default:
+					PerformLight();
+					break;
+
+			}
+		}
+
+		[Foreign(Language.ObjC)]
+		static extern(iOS) void PerformSoft()
+		@{
+		#if defined(__IPHONE_10_0) && __IPHONE_OS_VERSION_MAX_ALLOWED >= __IPHONE_10_0
+			dispatch_async(dispatch_get_main_queue(), ^{
+				UIImpactFeedbackGenerator * feedback = [[UIImpactFeedbackGenerator alloc] initWithStyle:UIImpactFeedbackStyleSoft];
+				[feedback impactOccurred];
+			});
+		#else
+			AudioServicesPlayAlertSound(kSystemSoundID_Vibrate);
+		#endif
+		@}
+
+		[Foreign(Language.ObjC)]
+		static extern(iOS) void PerformRigid()
+		@{
+		#if defined(__IPHONE_10_0) && __IPHONE_OS_VERSION_MAX_ALLOWED >= __IPHONE_10_0
+			dispatch_async(dispatch_get_main_queue(), ^{
+				UIImpactFeedbackGenerator * feedback = [[UIImpactFeedbackGenerator alloc] initWithStyle:UIImpactFeedbackStyleRigid];
+				[feedback impactOccurred];
+			});
+		#else
+			AudioServicesPlayAlertSound(kSystemSoundID_Vibrate);
+		#endif
+		@}
+
+		[Foreign(Language.ObjC)]
+		static extern(iOS) void PerformLight()
+		@{
+		#if defined(__IPHONE_10_0) && __IPHONE_OS_VERSION_MAX_ALLOWED >= __IPHONE_10_0
+			dispatch_async(dispatch_get_main_queue(), ^{
+				UIImpactFeedbackGenerator * feedback = [[UIImpactFeedbackGenerator alloc] initWithStyle:UIImpactFeedbackStyleLight];
+				[feedback impactOccurred];
+			});
+		#else
+			AudioServicesPlayAlertSound(kSystemSoundID_Vibrate);
+		#endif
+		@}
+
+		[Foreign(Language.ObjC)]
+		static extern(iOS) void PerformMedium()
+		@{
+		#if defined(__IPHONE_10_0) && __IPHONE_OS_VERSION_MAX_ALLOWED >= __IPHONE_10_0
+			dispatch_async(dispatch_get_main_queue(), ^{
+				UIImpactFeedbackGenerator * feedback = [[UIImpactFeedbackGenerator alloc] initWithStyle:UIImpactFeedbackStyleLight];
```

Comment:
This code looks identical to `PerformLight()`, do you mean `initWithStyle:UIImpactFeedbackStyleMedium` ?

Reply:
Ops! sorry about that, the evil of copy and paste, fix on the way